crates/router/src/route/segment.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::cmp::Ordering;
+
+#[derive(PartialEq, Eq, Debug)]
```

Comment:
```suggestion
#[derive(PartialEq, Eq, PartialOrd, Debug)]
```

Given we implement `Ord`, can we just derive `PartialOrd`, instead of having another block that calls to `Ord`?